### ISSUE TITLE
Question: should the CSS for styling hierarchical relationships be local or global?

### ISSUE THREAD

**Issue by matentzn** [EBISPOT/OLS#304]
It would be great if we could, similar to the cute blue (P) for part of, see other relations declared as hierarchical with a letter. I am a bit unsure whether this config should be local or global. 

| | Pro | Con |
| -- | -- | -- |
| Global | Standard look and feel, lower cognitive demand, P=partof, d=derivesfrom, etc etc; easier to implement and maintain for OLS devs |  letter space will exhaust quicker, conflicts, OLS devs have to maintain list  |
| Local (by ontology config) | Letter space does not exhaust; OLS devs do not need to property-letter mapping | Non-standard look and feel, cognitive demand to understand that "now 'p' means polar opposiste, not part of" |

In any case, I advocate a global solution. Would this be something you could consider as an enhancement?

**Comment by henrietteharmse** [comment]
Global is the only way that can be supported currently and is preferable to ensure ontologies in OLS are all rendered in a similar fashion.

For which property labels we want to add this is something that also require careful thought as this will not scale.

**Comment by matentzn** [comment]
Agreed then. :) How do we do this? Should we just suggest properties? I would suggest to definitely use the most important GO ones; in my current use case I need 'derived from'.. Where should we collect these? Here?

**Comment by henrietteharmse** [comment]
I think we should really just target 1, may be 2, properties for now.  We can role that out and then give some time that we can get feedback from users. Then we can consider some more. I am weary of large scale role out of this. Unintended side-effects of this can be the following:
(1) Because not all relations can be adorned by similar icons, those without icons might become more difficult to notice.
(2) If we have too many icons it may obfuscate information rather than highlight it to the point that even nodes with icons can be missed.
(3) If an ontology is using a property, but the property is not configured as a hierarchical property in OLS for that ontology, it will not show up.

As to which properties to target I think we must consider the following:
(1) These properties must be of particular interest to users. @matentzn here you insights can be very useful.
(2) The properties to target must be properties that are widely used across the various ontologies in OLS to ensure many ontologies will benefit from this.  That does not necessarily mean that a specific ontology should use it a lot - if a lot of ontologies each just use it a handful of times that is fine.  I should be able to write a cypher query to determine how prevalent a property is in OLS.

It might be a good idea to suggest potential properties in a Google doc with some motivation for or against it.

@simonjupp Do you perhaps have any thoughts on this?

**Comment by matentzn** [comment]
Awesome, great you are tackling this. 

1) There is no doubt about of 'part of' (P), and no doubt that 'has part' should NOT be used.

> (1) Because not all relations can be adorned by similar icons, those without icons might become more difficult to notice.

Right now, afaics, there is simply no icon, which makes it simply impossible to distinguish. Examples:

[Egypt in africa --> location in North Africa](https://www.ebi.ac.uk/ols/ontologies/gaz/terms?iri=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FGAZ_00000561&viewMode=All&siblings=false) appears nested, because GAZ decided located in is a hierarchical property. So the confusion is already 100%, and a green (L) would help I think.

> (2) If we have too many icons it may obfuscate information rather than highlight it to the point that even nodes with icons can be missed.

Again, as long as we permit arbitray hierachical properties, I think not having in icon is always worse than having one.. Alternatively we can add some global restrictions on which properties are allowed to be hierarchical in the first place..

> (3) If an ontology is using a property, but the property is not configured as a hierarchical property in OLS for that ontology, it will not show up.

Good point! 

I think the best empirical source of truth is to look at the OLS config and extraction a JSON like this:

```
- ontologies
   - id: oba
   - hierarchical:
      - RO:001
      - RO:002
```

And then simply count which are the most frequently used hierarchical properties. This should guide which relation to tackle first in my opinion!

What do you think?

**Comment by simonjupp** [comment]
Just to clarify that in OLS right now config happens at a per ontology level. It is globally assumed that `obo:part_of` is always hierarchical, but you can set other hierarchy properties at a per ontology level. For some `developed_from` and I think `located_in` for GAZ are set. 

If hierarchical properties are set, the tree will render them, but only shows an icon if it `part_of`. It would be god to get an icon set for the common properties.
